File: Fenicia.Module.Basic/Domains/Employee/Responses/GetAllEmployeeResponse.cs
using Fenicia.Module.Basic.Domains.Customer.Responses;

namespace Fenicia.Module.Basic.Domains.Employee.Responses;

/// <summary>
///     Response model for an employee in the list view.
///     Contains employee information including person and position details.
/// </summary>
public record GetAllEmployeeResponse(
    Guid Id, 
    Guid PositionId, 
    Guid PersonId, 
    string Name, 
    string? Email, 
    string? PhoneNumber, 
    string? Document, 
    string? PositionName,
    AddressResponse? Address);

File: Src/fenicia-web/src/types/basic/employee/get-all-employee-response.ts
import { AddressResponse } from "../address/address-response";

export type GetAllEmployeeResponse = {
    id: string;
    positionId: string;
    personId: string;
    name: string;
    email?: string;
    phoneNumber?: string;
    document?: string;
    positionName?: string;
    address?: AddressResponse;
};
